base/dates/periods.jl

```diff
@@ -166,10 +166,44 @@ type CompoundPeriod <: AbstractTime
                     break
                 end
             end
-            return new(reverse!(pc))
+            p = reverse(pc)
+            n = length(p)
         else
             return new(resize!(p, n))
         end
+        # canonicalize Period values such that they are all either
+        # positive or negative.
```

Comment:
I don't think this is actually possible.  e.g. consider `Year(-1) + Day(1)`.  Since `Year` is not a fixed number of days, it is not possible to make it all positive all negative.